integrations/db/src/utils/sync-game-state.ts

```diff
@@ -18,23 +18,44 @@ import {
 import { dbFieldsFromCharacter, characterFromDbFields } from "./transform";
 import type { CharacterId, AreaId } from "@mp/game-shared";
 import type { DrizzleClient } from "./client";
+import { Shape, ShapeStream } from "@electric-sql/client";
+import { startAsyncInterval, TimeSpan } from "@mp/time";
 
 export interface SyncGameStateOptions {
   area: AreaResource;
   state: GameState;
   actorModels: ActorModelLookup;
   logger: Logger;
   markToResendFullState: (characterId: CharacterId) => void;
+  electricUrl?: string;
+}
+
+export interface SyncGameStateSession {
+  /**
+   * Save current game state to the database
+   */
+  save: () => Promise<void>;
+  /**
+   * Stop the sync session and clean up resources
+   */
+  stop: () => void;
+  /**
+   * Forcefully flushes any pending sync operations to the database.
+   */
+  flush: () => void;
```

Comment:
flush should take an optional Character instance as argument. it doesn't need to be used for anything right now, but it needs to be in the signature.